providers/sendgrid/src/lib/sendgrid.provider.ts

```diff
@@ -25,7 +27,40 @@ export class SendgridEmailProvider implements IEmailProvider {
   async sendMessage(
     options: IEmailOptions
   ): Promise<ISendMessageSuccessResponse> {
-    const response = await this.sendgridMail.send({
+    const mailData = this.createMailData(options);
+    const response = await this.sendgridMail.send(mailData);
+
+    return {
+      id: response[0]?.headers['x-message-id'],
+      date: response[0]?.headers?.date,
+    };
+  }
+
+  async checkIntegration(
+    options: IEmailOptions
+  ): Promise<ICheckIntegrationResponse> {
+    try {
+      const mailData = this.createMailData(options);
+
+      const response = await this.sendgridMail.send(mailData);
+
+      if (response[0]?.statusCode === 202) {
+        return {
+          success: true,
+          message: 'Integration Successful',
+        };
+      }
+    } catch (error) {
+      return {
+        success: false,
+        message:
```

Comment:
Looks like here we are discarding the provider message at all, could we just return the Enum of our internal error on one field and also return the provider full text in case it exists? I think otherwise we might mask important information from the user. Wdyt? @BiswaViraj

Reply:
Providers' message is returned when none of the messages our Enum matches

`mapResponse(error?.code) || error?.response?.body?.errors[0]?.message,` handles it

Reply:
What I'm suggesting here, is to **always** return the provider full message and in another field our mapped enum error in case we want to handle it differently. Wdyt?

Reply:
I think that would be good.

`js
return {
 success: false,
 message: provider's response message,
 type: Mapped Enum(INVALID_EMAIL, BAD_CREDENTIALS.....)
`
Does this format look good? wdyt?

Reply:
Perfect! Would maybe just adjust the `type` to be `code`